lib/minitest/junit.rb

```diff
@@ -24,15 +26,24 @@ def record(result)
       end
 
       def report
-        @io.puts '<testsuite>'
-        @results.each { |result| @io.puts format(result) }
-        @io.puts '</testsuite>'
+        xml = Builder::XmlMarkup.new(:indent => 2)
+        xml.testsuite(name: 'minitest',
+                      timestamp: Time.now.iso8601,
+                      hostname: Socket.gethostname,
```

Comment:
I suggest moving these values to the options object in the constructor to avoid direct dependencies here (and we can make things more testable)

Reply:
Agreed, moved to constructor.